import org.junit.Test;
import static org.junit.Assert.*;

public class CalculatorTest {
    
    private final Calculator calculator = new Calculator();
    
    @Test
    public void testMultiply() {
        assertEquals(0, calculator.multiply(0, 5));
        assertEquals(10, calculator.multiply(2, 5));
        assertEquals(-15, calculator.multiply(3, -5));
        assertEquals(100, calculator.multiply(10, 10));
    }
    
    @Test
    public void testDivide() {
        assertEquals(2, calculator.divide(10, 5));
        assertEquals(-4, calculator.divide(20, -5));
        assertEquals(0, calculator.divide(0, 100));
    }
    
    @Test(expected = ArithmeticException.class)
    public void testDivideByZero() {
        calculator.divide(10, 0);
    }
}
